src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -102,4 +87,49 @@ public int size() {
     public boolean isEmpty() {
         return size == 0;
     }
+
+    private void rangeCheckForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
```

Comment:
I would add a little more information in the message, something like:
"Invalid index outside"
                + " the dimension of the array, index - " + index + System.lineSeparator()
                + "array size - " + size);